subsys/bluetooth/host/conn.c

```diff
@@ -1895,6 +1922,36 @@ int bt_conn_get_info(const struct bt_conn *conn, struct bt_conn_info *info)
 	return -EINVAL;
 }
 
+#if defined(CONFIG_BT_REMOTE_VERSION)
+int hci_le_read_remote_version(struct bt_conn *conn)
+{
+	int err;
+	struct bt_hci_cp_read_remote_version_info *cp;
+	struct net_buf *buf;
+
+	k_sem_init(&conn->rv.sem, 0, 1);
+
+	buf = bt_hci_cmd_create(BT_HCI_OP_READ_REMOTE_VERSION_INFO,
+				sizeof(*cp));
+	if (!buf) {
+		k_sem_give(&conn->rv.sem);
+		return -ENOBUFS;
+	}
+
+	cp = net_buf_add(buf, sizeof(*cp));
+	cp->handle = sys_cpu_to_le16(conn->handle);
+	atomic_set_bit(conn->flags, BT_CONN_PENDING_VERSION_INFO);
+	err = bt_hci_cmd_send(BT_HCI_OP_READ_REMOTE_VERSION_INFO, buf);
```

Comment:
Use `bt_hci_cmd_send_sync()` here - that way you wait for the Command Status HCI event and catch a potential error there (you'll get a non-zero return in that case). With the current PR if the HCI command would fail with a Command Status you'd still end up waiting the full HCI_CMD_TIMEOUT, which is not correct.